src/main/kotlin/com/game/catbackend/domain/services/LobbyService.kt

```diff
@@ -2,14 +2,40 @@ package com.game.catbackend.domain.services
 
 import com.game.catbackend.infra.repositories.LobbyRepository
 import com.game.catbackend.domain.entities.Lobby
+import com.game.catbackend.domain.enums.Status
+import com.game.catbackend.domain.exceptions.CatGameLobbyFullException
+import com.game.catbackend.domain.exceptions.CatGameLobbyNotFoundException
+import com.game.catbackend.domain.exceptions.CatGameLobbyStatusException
+import com.game.catbackend.domain.exceptions.CatGamePlayerAlreadyExistsException
+import com.game.catbackend.domain.services.PlayerService
 import org.springframework.stereotype.Service
 import java.util.*
 
 @Service
-class LobbyService(val lobbyRepository : LobbyRepository) {
+class LobbyService(val lobbyRepository : LobbyRepository, val playerService: PlayerService) {
 
     fun get(id: Long): Optional<Lobby> {
         return lobbyRepository.findById(id)
     }
 
+    fun joinLobby(roomId: UUID, playerName: String) {
+        val lobbyId = lobbyRepository.findIdByRoomId(roomId) ?: throw CatGameLobbyNotFoundException("Lobby does not exist.")
+        println(lobbyId)
+        val lobby = lobbyRepository.findById(lobbyId).orElseThrow { CatGameLobbyNotFoundException("Lobby does not exist.") }
```

Comment:
Podemos buscar o lobby inteiro de uma vez. Não precisamos bater no banco duas vezes
```suggestion
        val lobby = lobbyRepository.findByRoomId(roomId).orElseThrow { CatGameLobbyNotFoundException("Lobby does not exist.") }
```